### ISSUE TITLE
Train Muiti Obeject

### ISSUE THREAD

**Issue by qaz3014** [davisking/dlib#1231]
Hi Davis, 

I've tried your dnn_mmod_train_find_cars_ex and successfully trained a 2-types object detector. It's really powerful, great work dlib.

Recently I want to train a 21-types object detector. But the result is poor, the training loss is very high (about 1.2) and it can only classify 1 type of object. I've seem [#1191](https://github.com/davisking/dlib/issues/1191) and add my layer to 33.
`using net_type = dlib::loss_mmod<con<1, 9, 9, 1, 1, rcon5<rcon5<rcon5<rcon5<rcon5<rcon5<rcon5<downsampler<input_rgb_image_pyramid<pyramid_down<6>>>>>>>>>>>>`

But it still doesn't work. After checking my training data is correct, I divide my class to 9-types and 12-types by setting other labels to ignore and train them separately.
`if (box.label == "type_1")  u.ignore = true;`

Both results are good but I can't figure out why it doesn't work while I train them together. Have any idea why this could happened?

**Comment by davisking** [comment]
The network apparently isn't powerful enough to represent all the classes.
You need more layers or filters.  You have to experiment with these things
to see what works.

**Comment by qaz3014** [comment]
Thanks for the reply! I'll try to make it deeper and see how it works.

**Comment by qaz3014** [comment]
I've tried 45, 60, 75, 90 layers and the result seems to be the same. Now I'm trying 120 layers. Do you think adding more layer could help or I should try other models?

**Comment by davisking** [comment]
You probably have enough layers and there is something else wrong with what
you are doing.
